pkg/sm/sm.go

```diff
@@ -110,23 +110,31 @@ func New(ctx context.Context, cancel context.CancelFunc, env env.Environment) *S
 	log.C(ctx).Info("Setting up Service Manager storage...")
 
 	waitGroup := &sync.WaitGroup{}
-	smStorage := &postgres.PostgresStorage{}
-	if err := storage.OpenWithSafeTermination(ctx, smStorage, cfg.Storage, waitGroup); err != nil {
-		panic(fmt.Sprintf("error opening storage: %s", err))
+	encrypter := &security.AESEncrypter{}
+
+	smStorage := &postgres.Storage{
+		ConnectFunc: func(driver string, url string) (*sql.DB, error) {
+			return sql.Open(driver, url)
+		},
+		Encrypter: encrypter,
 	}
 
-	securityStorage := smStorage.Security()
-	if err = initializeSecureStorage(ctx, securityStorage); err != nil {
-		panic(fmt.Sprintf("error initialzing secure storage: %v", err))
+	// Initialize the storage with graceful termination
+	if err := storage.InitializeWithSafeTermination(ctx, smStorage, cfg.Storage, waitGroup); err != nil {
+		panic(fmt.Sprintf("error opening storage: %s", err))
 	}
 
-	encrypter := &security.TwoLayerEncrypter{
-		Fetcher: securityStorage.Fetcher(),
+	// Decorate the storage with credentials ecryption/decryption
+	encryptingRepository, err := storage.NewEncryptingRepository(ctx, smStorage, encrypter)
```

Comment:
What if I create a `NewEcnryptingRepository` with uninitialized storage?

Reply:
it will panic during setting up of the encryption key

Reply:
That's my concern. You need to know to do something hidden before constructing an object. That's what my proposal for hiding all of this behind a configuration. Like with flags what to enable - "I want a repository with interceptors and with encryption" would return a ready to use `storage.InterceptableTransactionalRepository`

Reply:
I agree that requiring the hidden Opening before creating the object should be adjusted but I am a bit concerned about allowing switching encryption on and off - imagine SM is started in ecrypt: off, some data is stored, then its restarted with encrypt: on  and some data is store again ... and stuff will start failing  as now we have both encrypted and unecrypted data